File: hengwen-web/src/types/report.ts
export type Severity = "error" | "warning" | "info";

export type Verdict = "pass" | "pending" | "fail";

export interface Issue {
  id: string;
  severity: Severity;
  title: string;
  location: string;
  summary: string;
  original: string;
  suggestion: string;
}

export interface Report {
  id: string;
  filename: string;
  fileType: ".docx" | ".pdf" | ".md";
  checkedAt: string;
  wordCount: number;
  score: number;
  verdict: Verdict;
  issues: Issue[];
}

export interface AnalysisTask {
  id: string;
  filename: string;
  fileType: Report["fileType"];
  stageIndex: number;
  progress: number;
  startedAt: string;
}

export interface Settings {
  orgName: string;
  standard: string;
  checkFormat: boolean;
  checkCitation: boolean;
  checkPlagiarism: boolean;
  autoReport: boolean;
}

File: hengwen-api/src/hengwen_api/schemas/report.py
from datetime import datetime
from typing import Literal

from pydantic import Field

from hengwen_api.schemas.common import CamelModel

FileType = Literal[".docx", ".pdf", ".md"]


class IssueResponse(CamelModel):
    id: str
    severity: Literal["error", "warning", "info"]
    title: str
    location: str
    summary: str
    original: str
    suggestion: str


class ReportResponse(CamelModel):
    id: str
    filename: str
    file_type: FileType
    checked_at: datetime
    word_count: int
    score: int
    verdict: Literal["pass", "pending", "fail"]
    issues: list[IssueResponse]


class ReportPageResponse(CamelModel):
    items: list[ReportResponse]
    page: int
    page_size: int
    total: int


class ReportPagination(CamelModel):
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)
